tree/ntupledraw/v7/src/RDrawStorage.cxx

```diff
@@ -0,0 +1,601 @@
+   std::string elementSizeOnStorage = "Element Size On Storage:" + std::string(11, ' ') +
+                                      std::to_string(8 * fLocator.fBytesOnStorage / fNElements) + " bits";
+   latex.DrawLatex(0.01, 0.45, elementSizeOnStorage.c_str());
+   std::string pageSize =
+      "Page Size On Disk:" + std::string(22, ' ') + std::to_string(fNElements * fElementSizeOnDisk / 8) + " bytes";
+   latex.DrawLatex(0.01, 0.40, pageSize.c_str());
+   std::string pageSizeStorage =
+      "Page Size On Storage:" + std::string(16, ' ') + std::to_string(fLocator.fBytesOnStorage) + " bytes";
+   latex.DrawLatex(0.01, 0.35, pageSizeStorage.c_str());
+   std::string globalRange = "Global Page Range:" + std::string(21, ' ') + std::to_string(fGlobalRangeStart) + " - " +
+                             std::to_string(fGlobalRangeStart + fNElements - 1);
+   latex.DrawLatex(0.01, 0.30, globalRange.c_str());
+   std::string clusterRange = "Cluster Page Range:" + std::string(20, ' ') + std::to_string(fClusterRangeStart) +
+                              " - " + std::to_string(fClusterRangeStart + fNElements - 1);
+   latex.DrawLatex(0.01, 0.25, clusterRange.c_str());
+   std::size_t totalNumBytes = fParent->GetTotalNumBytes();
+   std::size_t scalingFactorOfAxis = fParent->GetScalingFactorOfAxis();
+   std::string locationString = "Location in Storage:" + std::string(20, ' ') +
+                                std::to_string(static_cast<std::size_t>(GetX1() * scalingFactorOfAxis)) + " / " +
+                                std::to_string(totalNumBytes) + " bytes";
+   latex.DrawLatex(0.01, 0.20, locationString.c_str());
+
+   fParent->fCanvasPtrs.back()->Update();
+}
+
+ClassImp(ROOT::Experimental::Detail::RPageBox)
+
+   // ------------------------------ RDrawStorage ------------------------------
+
+   ROOT::Experimental::Detail::RPageBox::RPageBox(double x1, double y1, double x2, double y2, std::string fieldName,
+                                                  std::string fieldType, DescriptorId_t fieldId,
+                                                  DescriptorId_t columnId, DescriptorId_t clusterId,
+                                                  EColumnType columnType, ClusterSize_t::ValueType nElements,
+                                                  NTupleSize_t globalRangeStart, NTupleSize_t clusterRangeStart,
+                                                  RClusterDescriptor::RLocator locator, RDrawStorage *parent,
+                                                  std::size_t pageBoxId)
+   : TBox(x1, y1, x2, y2), fFieldName{fieldName}, fFieldType{fieldType}, fFieldId{fieldId}, fColumnId{columnId},
+     fClusterId{clusterId}, fNElements{nElements}, fGlobalRangeStart{globalRangeStart},
+     fClusterRangeStart{clusterRangeStart}, fLocator{locator}, fParent{parent}, fPageBoxId{pageBoxId}
+{
+   switch (columnType) {
+   case EColumnType::kIndex:
+      fColumnType = "Index";
```

Comment:
I think we should put that information in `RColumnModel.hxx`.  There can be static arrays with the names and on disk sizes, so that we can use something like `kColumnNames[static_cast<int>(columnType)]`

Reply:
Good idea, I implemented it.